doboto/Endpoint.py

```diff
@@ -80,3 +80,51 @@ def pages(self, request_url, expect, params=None):
                 next_url = None
 
         return items
+
+    def action_result(self, action, wait, poll, timeout):
+        """
+        General action result processor for waiting
+        """
+
+        start_time = time.time()
+
+        while wait and action["status"] == "in-progress":
+
+            time.sleep(poll)
+            try:
+                action = self.do.action.info(action["id"])
+            except:
+                pass
```

Comment:
That's cool and all, but if we're going to do that, we should specify the subset of exceptions that could do that rather than just blanket throwing away all exceptions.  You can things like this:

```python

exceptions_we_are_ok_with = (ExceptionClass1, ExceptionClass2, ExceptionClass3 ....)


try:
    do_a_thing_that_could_throw_an_exception
except exceptions_we_are_ok_with:
    pass

# All other exceptions bubble up

```